### PR TITLE
tts add voice choose

### PR DESCRIPTION
**Pull request by charli117**

### CODE REVIEW

File: api/core/model_runtime/model_providers/openai/tts/tts-1-hd.yaml [hunk 1]
```diff
@@ -1,7 +1,31 @@
-model: tts-1-hd
+model: tts-1
 model_type: tts
 model_properties:
   default_voice: 'alloy'
+  voices:
+    - mode: 'alloy'
+      name: 'Alloy'
+      language: ['zh-CN-CN', 'en-US']
```
Comment (takatost) [inline api/core/model_runtime/model_providers/openai/tts/tts-1-hd.yaml]: looks like there's a mistake here.

File: api/core/model_runtime/model_providers/openai/tts/tts-1.yaml [hunk 2]
```diff
@@ -2,6 +2,30 @@ model: tts-1
 model_type: tts
 model_properties:
   default_voice: 'alloy'
+  voices:
+    - mode: 'alloy'
+      name: 'Alloy'
+      language: ['zh-CN-CN', 'en-US']
```
Comment (takatost) [inline api/core/model_runtime/model_providers/openai/tts/tts-1.yaml]: there's a mistake too.

**Review by takatost — CHANGES_REQUESTED** [review]
Thank you so much for your contribution! Btw, could you pls help us update the contribution doc for `tts` as well?

**Comment by charli117** [comment]
> Thank you so much for your contribution! Btw, could you pls help us update the contribution doc for `tts` as well?

It has been fixed, and update tts doc

**Review by takatost — CHANGES_REQUESTED** [review]
image
App that has been configured with `text-to-speech` will throw an error due to failed verification. Maybe we can add a default value?

**Comment by takatost** [comment]
```
ERROR:root:internal server error, 'voice'.
Traceback (most recent call last):
  File "/app/api/controllers/console/app/audio.py", line 98, in post
    voice=app_model.app_model_config.text_to_speech_dict['voice'],
KeyError: 'voice'
ERROR:app:Exception on /console/api/apps/3b950e1e-7f8a-4d22-b13c-12710e3bf6a6/text-to-audio [POST]
Traceback (most recent call last):
  File "/app/api/controllers/console/app/audio.py", line 98, in post
    voice=app_model.app_model_config.text_to_speech_dict['voice'],
KeyError: 'voice'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 870, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 855, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
  File "/usr/local/lib/python3.10/site-packages/flask_restful/__init__.py", line 489, in wrapper
    resp = resource(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/flask/views.py", line 110, in view
    return current_app.ensure_sync(self.dispatch_request)(**kwargs)  # type: ignore[no-any-return]
  File "/usr/local/lib/python3.10/site-packages/flask_restful/__init__.py", line 604, in dispatch_request
    resp = meth(*args, **kwargs)
  File "/app/api/controllers/console/setup.py", line 84, in decorated
    return view(*args, **kwargs)
  File "/app/api/libs/login.py", line 91, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
  File "/app/api/controllers/console/wraps.py", line 21, in decorated
    return view(*args, **kwargs)
  File "/app/api/controllers/console/app/audio.py", line 126, in post
    raise InternalServerError()
werkzeug.exceptions.InternalServerError: 500 Internal Server Error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.
```

And after completing the tts configuration, when debugging on the orchestration page, it will throw an Internal Server Error.

**Comment by charli117** [comment]
> ```
> ERROR:root:internal server error, 'voice'.
> Traceback (most recent call last):
>   File "/app/api/controllers/console/app/audio.py", line 98, in post
>     voice=app_model.app_model_config.text_to_speech_dict['voice'],
> KeyError: 'voice'
> ERROR:app:Exception on /console/api/apps/3b950e1e-7f8a-4d22-b13c-12710e3bf6a6/text-to-audio [POST]
> Traceback (most recent call last):
>   File "/app/api/controllers/console/app/audio.py", line 98, in post
>     voice=app_model.app_model_config.text_to_speech_dict['voice'],
> KeyError: 'voice'
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 870, in full_dispatch_request
>     rv = self.dispatch_request()
>   File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 855, in dispatch_request
>     return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
>   File "/usr/local/lib/python3.10/site-packages/flask_restful/__init__.py", line 489, in wrapper
>     resp = resource(*args, **kwargs)
>   File "/usr/local/lib/python3.10/site-packages/flask/views.py", line 110, in view
>     return current_app.ensure_sync(self.dispatch_request)(**kwargs)  # type: ignore[no-any-return]
>   File "/usr/local/lib/python3.10/site-packages/flask_restful/__init__.py", line 604, in dispatch_request
>     resp = meth(*args, **kwargs)
>   File "/app/api/controllers/console/setup.py", line 84, in decorated
>     return view(*args, **kwargs)
>   File "/app/api/libs/login.py", line 91, in decorated_view
>     return current_app.ensure_sync(func)(*args, **kwargs)
>   File "/app/api/controllers/console/wraps.py", line 21, in decorated
>     return view(*args, **kwargs)
>   File "/app/api/controllers/console/app/audio.py", line 126, in post
>     raise InternalServerError()
> werkzeug.exceptions.InternalServerError: 500 Internal Server Error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.
> ```
> 
> And after completing the tts configuration, when debugging on the orchestration page, it will throw an Internal Server Error.

@takatost fixed initialization null error, this is because the voice is not configured, resulting in the absence of two textToSpeechConfig property fields, voice and lanuage

**Comment by takatost** [comment]
> > ```
> > ERROR:root:internal server error, 'voice'.
> > Traceback (most recent call last):
> >   File "/app/api/controllers/console/app/audio.py", line 98, in post
> >     voice=app_model.app_model_config.text_to_speech_dict['voice'],
> > KeyError: 'voice'
> > ERROR:app:Exception on /console/api/apps/3b950e1e-7f8a-4d22-b13c-12710e3bf6a6/text-to-audio [POST]
> > Traceback (most recent call last):
> >   File "/app/api/controllers/console/app/audio.py", line 98, in post
> >     voice=app_model.app_model_config.text_to_speech_dict['voice'],
> > KeyError: 'voice'
> > 
> > During handling of the above exception, another exception occurred:
> > 
> > Traceback (most recent call last):
> >   File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 870, in full_dispatch_request
> >     rv = self.dispatch_request()
> >   File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 855, in dispatch_request
> >     return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
> >   File "/usr/local/lib/python3.10/site-packages/flask_restful/__init__.py", line 489, in wrapper
> >     resp = resource(*args, **kwargs)
> >   File "/usr/local/lib/python3.10/site-packages/flask/views.py", line 110, in view
> >     return current_app.ensure_sync(self.dispatch_request)(**kwargs)  # type: ignore[no-any-return]
> >   File "/usr/local/lib/python3.10/site-packages/flask_restful/__init__.py", line 604, in dispatch_request
> >     resp = meth(*args, **kwargs)
> >   File "/app/api/controllers/console/setup.py", line 84, in decorated
> >     return view(*args, **kwargs)
> >   File "/app/api/libs/login.py", line 91, in decorated_view
> >     return current_app.ensure_sync(func)(*args, **kwargs)
> >   File "/app/api/controllers/console/wraps.py", line 21, in decorated
> >     return view(*args, **kwargs)
> >   File "/app/api/controllers/console/app/audio.py", line 126, in post
> >     raise InternalServerError()
> > werkzeug.exceptions.InternalServerError: 500 Internal Server Error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > And after completing the tts configuration, when debugging on the orchestration page, it will throw an Internal Server Error.
> 
> @takatost fixed initialization null error, this is because the voice is not configured, resulting in the absence of two textToSpeechConfig property fields, voice and lanuage

This issue has been resolved and can now play perfectly. 👍🏻
However, the previous `Validation exception` still exists. If this PR is merged and deployed, it will cause the existing app to throw an error during voice playing. Developers will need to manually set the voice and language for it to run properly.

**Comment by charli117** [comment]
> > > ```
> > > ERROR:root:internal server error, 'voice'.
> > > Traceback (most recent call last):
> > >   File "/app/api/controllers/console/app/audio.py", line 98, in post
> > >     voice=app_model.app_model_config.text_to_speech_dict['voice'],
> > > KeyError: 'voice'
> > > ERROR:app:Exception on /console/api/apps/3b950e1e-7f8a-4d22-b13c-12710e3bf6a6/text-to-audio [POST]
> > > Traceback (most recent call last):
> > >   File "/app/api/controllers/console/app/audio.py", line 98, in post
> > >     voice=app_model.app_model_config.text_to_speech_dict['voice'],
> > > KeyError: 'voice'
> > > 
> > > During handling of the above exception, another exception occurred:
> > > 
> > > Traceback (most recent call last):
> > >   File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 870, in full_dispatch_request
> > >     rv = self.dispatch_request()
> > >   File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 855, in dispatch_request
> > >     return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
> > >   File "/usr/local/lib/python3.10/site-packages/flask_restful/__init__.py", line 489, in wrapper
> > >     resp = resource(*args, **kwargs)
> > >   File "/usr/local/lib/python3.10/site-packages/flask/views.py", line 110, in view
> > >     return current_app.ensure_sync(self.dispatch_request)(**kwargs)  # type: ignore[no-any-return]
> > >   File "/usr/local/lib/python3.10/site-packages/flask_restful/__init__.py", line 604, in dispatch_request
> > >     resp = meth(*args, **kwargs)
> > >   File "/app/api/controllers/console/setup.py", line 84, in decorated
> > >     return view(*args, **kwargs)
> > >   File "/app/api/libs/login.py", line 91, in decorated_view
> > >     return current_app.ensure_sync(func)(*args, **kwargs)
> > >   File "/app/api/controllers/console/wraps.py", line 21, in decorated
> > >     return view(*args, **kwargs)
> > >   File "/app/api/controllers/console/app/audio.py", line 126, in post
> > >     raise InternalServerError()
> > > werkzeug.exceptions.InternalServerError: 500 Internal Server Error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > 并且完成tts配置后，在编排页面调试时，会抛出Internal Server Error。
> > 
> > 
> > @takatost修复初始化为null的错误，这是因为没有配置语音，导致缺少voice和lanuage这两个textToSpeechConfig属性字段
> 
> 这个问题已经解决了，现在可以完美播放了。 👍🏻 不过，以前的`Validation exception`依然存在。如果合并并部署此 PR，将导致现有应用程序在语音播放时抛出错误。开发人员需要手动设置语音和语言才能正常运行。

@

**Comment by charli117** [comment]
> > > ```
> > > ERROR:root:internal server error, 'voice'.
> > > Traceback (most recent call last):
> > >   File "/app/api/controllers/console/app/audio.py", line 98, in post
> > >     voice=app_model.app_model_config.text_to_speech_dict['voice'],
> > > KeyError: 'voice'
> > > ERROR:app:Exception on /console/api/apps/3b950e1e-7f8a-4d22-b13c-12710e3bf6a6/text-to-audio [POST]
> > > Traceback (most recent call last):
> > >   File "/app/api/controllers/console/app/audio.py", line 98, in post
> > >     voice=app_model.app_model_config.text_to_speech_dict['voice'],
> > > KeyError: 'voice'
> > > 
> > > During handling of the above exception, another exception occurred:
> > > 
> > > Traceback (most recent call last):
> > >   File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 870, in full_dispatch_request
> > >     rv = self.dispatch_request()
> > >   File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 855, in dispatch_request
> > >     return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
> > >   File "/usr/local/lib/python3.10/site-packages/flask_restful/__init__.py", line 489, in wrapper
> > >     resp = resource(*args, **kwargs)
> > >   File "/usr/local/lib/python3.10/site-packages/flask/views.py", line 110, in view
> > >     return current_app.ensure_sync(self.dispatch_request)(**kwargs)  # type: ignore[no-any-return]
> > >   File "/usr/local/lib/python3.10/site-packages/flask_restful/__init__.py", line 604, in dispatch_request
> > >     resp = meth(*args, **kwargs)
> > >   File "/app/api/controllers/console/setup.py", line 84, in decorated
> > >     return view(*args, **kwargs)
> > >   File "/app/api/libs/login.py", line 91, in decorated_view
> > >     return current_app.ensure_sync(func)(*args, **kwargs)
> > >   File "/app/api/controllers/console/wraps.py", line 21, in decorated
> > >     return view(*args, **kwargs)
> > >   File "/app/api/controllers/console/app/audio.py", line 126, in post
> > >     raise InternalServerError()
> > > werkzeug.exceptions.InternalServerError: 500 Internal Server Error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > And after completing the tts configuration, when debugging on the orchestration page, it will throw an Internal Server Error.
> > 
> > 
> > @takatost fixed initialization null error, this is because the voice is not configured, resulting in the absence of two textToSpeechConfig property fields, voice and lanuage
> 
> This issue has been resolved and can now play perfectly. 👍🏻 However, the previous `Validation exception` still exists. If this PR is merged and deployed, it will cause the existing app to throw an error during voice playing. Developers will need to manually set the voice and language for it to run properly.

@takatost This bug has been fixed，adjusted the initial value of TextToSpeechEntity BaseModel

**Review by takatost — CHANGES_REQUESTED** [review]
image

Hey, I've run into a new issue. When I choose OpenAI's Onyx for English and most of the voices for Chinese, I'm facing some problems.

**Comment by charli117** [comment]
> image
> Hey, I've run into a new issue. When I choose OpenAI's Onyx for English and most of the voices for Chinese, I'm facing some problems.

@takatost This error is a bit strange, did you pull the openai tts yaml file in full? I'll take the type restrictions off. You try again

**Review by takatost — APPROVED** [review]
Testing is all good now! 🎉
Thank you so much for your contribution!

**Comment by takatost** [comment]
@charli117 Hey, I found a new issue. Even when I switch to the English interface, the language dropdown still shows Chinese. 🤔

**Pull request merged**